libraries/aws_iam_user.rb

```diff
@@ -15,29 +15,32 @@ class AwsIamUser < Inspec.resource(1)
   def initialize(
     opts,
     aws_user_provider = AwsIam::UserProvider.new,
+    aws_user_details_provider_factory = AwsIam::UserDetailsProviderFactory.new,
     access_key_factory = AwsIamAccessKeyFactory.new
   )
     @user = opts[:user]
     @user = aws_user_provider.user(opts[:name]) if @user.nil?
+    @aws_user_details_provider = aws_user_details_provider_factory.create(@user)
     @access_key_factory = access_key_factory
+    @aws_user_provider = aws_user_provider
```

Comment:
Do we need the `@user` or `@aws_user_provider` instance variables?  If not, please remove them.

Reply:
I suppose not, with this approach.

- [ ] Remove